File: Algorithms/Python3.x/461-Hamming_Distance.py
class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        dis = 0
        for i in bin(x ^ y):
            if i == '1': dis += 1
        return dis


class Solution(object):
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        return sum(x == '1' for x in bin(x ^ y))


# Runtime: 24 ms, faster than 99.64% of Python3 online submissions for Hamming Distance.
# Memory Usage: 12.7 MB, less than 100.00% of Python3 online submissions for Hamming Distance.
class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        return bin(x ^ y).count('1')

File: Algorithms/C++/461-Hamming_Distance.cpp
// Runtime: 4 ms, faster than 56.44% of C++ online submissions for Hamming Distance.
// Memory Usage: 8.3 MB, less than 40.91% of C++ online submissions for Hamming Distance.
class Solution {
public:
    int hammingDistance(int x, int y) {
        int cnt = 0, num = x ^ y;
        while (num > 0) {
            cnt += num & 1;
            num >>= 1;
        }
        return cnt;
    }
};
